File: webapp/src/types/userSettings.ts
// Mirrors UserSettingsRead / UserSettingsUpdate (backend/app/schemas/user_settings.py).
// Delivery *preferences* — distinct from the Notification feed itself (see
// types/notification.ts).
export interface UserSettings {
  // null = "use the global settings.REMINDER_LEAD_HOURS default".
  reminder_lead_hours: number | null
  notifications_enabled: boolean
  email_notifications_enabled: boolean
  push_notifications_enabled: boolean
}

// The backend applies this via exclude_unset, so an omitted key is left
// untouched — same convention as ContactUpdatePayload/DocumentUpdatePayload.
// An explicit `reminder_lead_hours: null` resets it to the global default.
export type UserSettingsUpdatePayload = Partial<UserSettings>

File: mobile/lib/features/settings/domain/user_settings.dart
/// Mirrors the backend's `UserSettingsRead` schema
/// (backend/app/schemas/user_settings.py) — delivery *preferences* for
/// interview reminders. A distinct concept from the in-app notification
/// feed itself (see features/notifications/domain/notification.dart).
class UserSettings {
  const UserSettings({
    required this.notificationsEnabled,
    required this.emailNotificationsEnabled,
    required this.pushNotificationsEnabled,
    this.reminderLeadHours,
  });

  /// Null = "use the backend's global default lead time".
  final int? reminderLeadHours;
  final bool notificationsEnabled;
  final bool emailNotificationsEnabled;
  final bool pushNotificationsEnabled;

  factory UserSettings.fromJson(Map<String, dynamic> json) {
    return UserSettings(
      reminderLeadHours: json['reminder_lead_hours'] as int?,
      notificationsEnabled: json['notifications_enabled'] as bool,
      emailNotificationsEnabled: json['email_notifications_enabled'] as bool,
      pushNotificationsEnabled: json['push_notifications_enabled'] as bool,
    );
  }
}
